src/epoch/initializers.rs

```diff
@@ -0,0 +1,409 @@
+    pub fn from_utc_seconds(seconds: f64) -> Self {
+        Self::from_utc_duration(seconds * Unit::Second)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided UTC days since 1900 January 01 at midnight
+    pub fn from_utc_days(days: f64) -> Self {
+        Self::from_utc_duration(days * Unit::Day)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since 1980 January 6 at midnight
+    pub fn from_gpst_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::GPST)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since 1980 January 6 at midnight
+    pub fn from_qzsst_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::QZSST)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since August 21st 1999 midnight
+    pub fn from_gst_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::GST)
+    }
+
+    #[must_use]
+    /// Initialize an Epoch from the provided duration since January 1st midnight
+    pub fn from_bdt_duration(duration: Duration) -> Self {
+        Self::from_duration(duration, TimeScale::BDT)
+    }
+
+    #[must_use]
+    pub fn from_mjd_tai(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::TAI)
+    }
+
+    pub fn from_mjd_in_time_scale(days: f64, time_scale: TimeScale) -> Self {
+        assert!(
+            days.is_finite(),
+            "Attempted to initialize Epoch with non finite number"
+        );
+        Self {
+            duration: (days - MJD_J1900) * Unit::Day,
+            time_scale,
+        }
+    }
+
+    #[must_use]
+    pub fn from_mjd_utc(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::UTC)
+    }
+    #[must_use]
+    pub fn from_mjd_gpst(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::GPST)
+    }
+    #[must_use]
+    pub fn from_mjd_qzsst(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::QZSST)
+    }
+    #[must_use]
+    pub fn from_mjd_gst(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::GST)
+    }
+    #[must_use]
+    pub fn from_mjd_bdt(days: f64) -> Self {
+        Self::from_mjd_in_time_scale(days, TimeScale::BDT)
+    }
+
+    #[must_use]
+    pub fn from_jde_tai(days: f64) -> Self {
+        Self::from_jde_in_time_scale(days, TimeScale::TAI)
+    }
+
+    fn from_jde_in_time_scale(days: f64, time_scale: TimeScale) -> Self {
+        assert!(
+            days.is_finite(),
+            "Attempted to initialize Epoch with non finite number"
+        );
+        Self {
+            duration: (days - MJD_J1900 - MJD_OFFSET) * Unit::Day,
+            time_scale,
+        }
+    }
+
+    #[must_use]
+    pub fn from_jde_utc(days: f64) -> Self {
+        Self::from_jde_in_time_scale(days, TimeScale::UTC)
```

Comment:
The `from_jde_utc` function has the same bug as in #302 , so I've done the same fix.